src/matlab2tikz.m

```diff
@@ -1366,6 +1367,18 @@ case guitypes()
         end
         str = join(m2t, str, '\\[1ex]');
         opts =  opts_add(opts, tikzKeyword, sprintf('{%s}', str));
+    
+        % set color of axis labels
+        color = object.Color;
+        [m2t, col] = getColor(m2t, handle, color, 'patch');
+        if strcmpi(labelKind,'title')
+          % TODO: color titles
+        else
+          opts = ...
+              opts_add(opts, ...
+              ['every axis ',lower(labelKind(1)),' label/.append style'], ...
```

Comment:
The `every X` seems strange, but that's how it's supposed to be. In any case (or as far as I understand and have some clarity), those `every X` statements are still contained within the `axis` (and `tikzpicture`) scope, so it shouldn't propagate further to other scopes.

I think it looks nicer without the `every`, so probably this excerpt from the manual helps a bit to pick an/any alternative:
<img width="897" alt="screen shot 2016-04-08 at 19 12 23" src="https://cloud.githubusercontent.com/assets/177360/14391707/efc4b41c-fdbd-11e5-9a4d-56db511610fc.png">